python/lsst/summit/utils/butlerUtils.py

```diff
@@ -478,10 +478,12 @@ def _assureDict(dataId: dict | dafButler.dimensions.DataCoordinate | dafButler.D
     """
     if isinstance(dataId, dict):
         return dataId
+    elif hasattr(dataId, "mapping"):  # dafButler.dimensions.DataCoordinate
+        return {str(k): v for k, v in dataId.mapping.items()}
     elif hasattr(dataId, "items"):  # dafButler.dimensions.DataCoordinate
         return {str(k): v for k, v in dataId.items()}  # str() required due to full names
-    elif hasattr(dataId, "dataId"):  # dafButler.DimensionRecord
-        return {str(k): v for k, v in dataId.dataId.items()}
+    elif hasattr(dataId, "dataId"):  # dafButler.dimensions.DimensionRecord
```

Comment:
Why did you add the `.dimensions` here? It's exported to the module level, so it wasn't wrong before.